Assets/Scripts/ExerciseManager.cs

```diff
@@ -60,22 +80,49 @@ public class ExerciseManager : Singleton<ExerciseManager> {
             curExercise.CleanUp();
             ToggleCurrentExercise(false);
             curExercisePos += 1;
-            if (curExercisePos == exerciseList.Count) {
+            if (curExercisePos == fileEntries.Length) {
                 InitiateFreePlay();
             }
             else {
                 LoggingManager.instance.UpdateLogColumn(curExcersieCol, curExercisePos.ToString());
-                curExercise = exerciseList[curExercisePos];
+                Debug.Log(curExercisePos);
+                SetUpCurExercise(curExercisePos);
                 ToggleCurrentExercise(true);
             }
         }
 
         private void InitiateFreePlay() {
             Debug.Log("Free play woould be initiated");
+            curExercise = exerciseList[0];
+            ToggleCurrentExercise(true);
         }
 
         private void ToggleCurrentExercise(bool desiredActiveState) {
             curExercise.gameObject.SetActive(desiredActiveState);
         }
+
+        private void SetupCodeBlockDictionary() {
+            codeBlockDictionary = new Dictionary<string, GameObject>();
+            codeBlockDictionary.Add("Print", Resources.Load<GameObject>(ResourcePathConstants.PrintCodeBlockPrefab));
```

Comment:
is there a way to just put it in a string array in ResourcePathConstants? this way people won't forget to add it here when a new codeblock is added as you already need to update resourcepathconstants when you make a new block